### PR TITLE
CORTX-33880: podcrash- memory limit converted from Gi, Pi, Ti or G, T, P to MiB

### PR DESCRIPTION
**Pull request by vaishnavirjoshi**
# Pull Request
## Problem Statement
- podcrash- memory limit converted from Gi, Pi, Ti or G, T, P to MiB
-  Jira id : [CORTX-33880](https://jts.seagate.com/browse/CORTX-33880)

##FIX
- Control pod in running after giving memory limit into Ki,  Ki, Gi, Pi, Mi, Ti, K, G, M, B, T, P

## Design
-   For Bug, Describe the fix here.
-   For Feature, Post the link for design

-   \[ \] Coding conventions are followed and code is consistent

-   \[ \] Tested by changing solution.yaml file and redeployed cortx. All pods are running. 
-   \[ \] Test Cases cover Happy Path, Non-Happy Path and Scalability
-   \[ \] Testing was performed with RPM
- 
image
image
image
image

-   \[ \] Interface change (if any) are documented
-   \[ \] Side effects on other features (deployment/upgrade)
-   \[ \] Dependencies on other component(s)

-   \[ \] JIRA number/GitHub Issue added to PR
-   \[ \] PR is self reviewed
-   \[ \] Jira and state/status is updated and JIRA is updated with PR link
-   \[ \] Check if the description is clear and explained

-   \[ \] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

File: csm/conf/configure.py [hunk 1]
```diff
@@ -361,8 +361,20 @@ def _mem_limit_to_int(mem: str) -> int:
...
-        return int(numbers_only)
+#         numbers_only = mem[:mem.find('M')]
+#         return int(numbers_only)
+        num = ''
+        unit = ''
+        for char in mem:
+            if ord(char) >= 48 and ord(char) <= 57:
+                num = num + char
+            elif (ord(char) >= 65 and ord(char) <= 90) or (ord(char) >= 97 and ord(char) <= 122):
+                unit += char
```
Comment (rohitkolapkar) [inline csm/conf/configure.py:372]: As per this logic if  mem="12b3c5b", then result will be something like below,
num = 1235
unit = bcb

This is not the expected result. 

Try to use regex here to separate out mem value and mem unit.

**Review by rohitkolapkar — CHANGES_REQUESTED** [review]
please handle codacy issues.

File: csm/conf/configure.py [hunk 2]
```diff
@@ -361,8 +361,20 @@ def _mem_limit_to_int(mem: str) -> int:
...
+
+        intVal = int(num)
+        convertFrom = unit
+        convertTo = 'M'
```
[the remark this answers is not in the document]
Comment (vaishnavirjoshi) [reply csm/conf/configure.py:376]: M is for mb and Mi is for Mib. So here default is set to Mb as per the requirements.

File: csm/conf/configure.py (hunk 1 above)
Comment (vaishnavirjoshi) [reply csm/conf/configure.py:372]: ok I'll add regex

File: csm/common/utility.py [hunk 3]
```diff
@@ -125,3 +125,11 @@ def load_csm_config_indices(conf):
             except (VError, KvError) as e:
                 Log.error(f"Unable to fetch the configurations from consul: {e}")
                 raise e
+                
+    @staticmethod
+    def conversion_mem_limit(intVal, convertFrom, convertTo):
+        convertToByte = {'B' : 1, 'K' : 1000, 'M' : 1000**2, 'G' : 1000**3, 'T' : 1000 ** 4, 'P' : 1000 ** 5,'Ki' : 1024, 'Mi' : 1024**2, 'Gi' : 1024**3, 'Ti' : 1024 ** 4, 'Pi' : 1024 ** 5}
```
[the remark this answers is not in the document]
Comment (vaishnavirjoshi) [reply csm/common/utility.py]: yes

**Comment by shailesh-vaidya** [comment]
Closing as an obsolete

**Pull request closed without merging**